plugins/woocommerce/src/Internal/Updates/BatchProcessingController.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * Controller to manage updates of `BatchProcessor` type. Trigger the `enqueue_processor` method to start a update, and it will manage rest of the things.
+ */
+
+namespace Automattic\WooCommerce\Internal\Updates;
+
+use Automattic\WooCommerce\DataBase\BatchProcessor;
+
+/**
+ * Class BatchProcessingController
+ *
+ * @package Automattic\WooCommerce\Internal\Updates.
+ */
+class BatchProcessingController {
+
+	const PENDING_PROCESS_OPTION_NAME = 'wc_pending_action_updates';
+	const CONTROLLER_CRON_NAME = 'wc_schedule_pending_updates';
+	const SINGLE_PROCESS_MIGRATION_ACTION = 'wc_run_action_update';
+	const ACTION_GROUP = 'wc_updates';
```

Comment:
Since we are moving to the concept of "batches", I suggest renaming the options and actions as folloes:

```suggestion
	const PENDING_PROCESS_OPTION_NAME = 'wc_pending_batch_updates';
	const CONTROLLER_CRON_NAME = 'wc_schedule_pending_batch_updates';
	const SINGLE_PROCESS_BATCH_ACTION = 'wc_run_batch_update';
	const ACTION_GROUP = 'wc_batch_updates';
```

Reply:
Thanks, added in https://github.com/woocommerce/woocommerce/pull/33233/commits/e922be012e20ff76dc1fc8357d357b573f8c4cd2#diff-8989ad7f50c6e63bf0ff83b5df17d72121166df76e16875ec985062b8ca740d5R17-R20